app/(main)/account/recovery/page.test.tsx

```diff
@@ -0,0 +1,152 @@
+    jest.clearAllMocks();
+  });
+
+  const fillFormAndSubmit = () => {
+    fireEvent.change(screen.getByTestId('password'), {
+      target: { value: 'newpassword123' },
+    });
+    fireEvent.change(screen.getByTestId('confirm-password'), {
+      target: { value: 'newpassword123' },
+    });
+    fireEvent.click(screen.getByTestId('continue-button'));
+  };
+
+  it('should render the reset password form when all search params are present', async () => {
+    render(<ResetPassword />);
+    await screen.findByText('Reset Your Password');
+    expect(screen.getByTestId('password')).toBeInTheDocument();
+    expect(screen.getByTestId('confirm-password')).toBeInTheDocument();
+  });
+
+  it('should redirect to login page when search params are missing', async () => {
+    mockUseSearchParams.mockReturnValue(null);
+    render(<ResetPassword />);
+    await waitFor(() => {
+      expect(mockRouter.push).toHaveBeenCalledWith('/login');
+    });
+  });
+
+  it('should submit the form and reset password successfully', async () => {
+    (resetPassword as jest.Mock).mockResolvedValue({});
+    render(<ResetPassword />);
+    fillFormAndSubmit();
+
+    await waitFor(() => {
+      expect(resetPassword).toHaveBeenCalledWith({
+        userId: '123',
+        token: 'abc',
+        password: 'newpassword123',
+      });
+      expect(mockToast).toHaveBeenCalledWith(
+        <Alert
+          variant={AlertVariants.Success}
+          message="Your password has been reset successfully! Please login with your new password."
+        />,
+      );
+    });
+
+    act(() => {
+      jest.advanceTimersByTime(2000);
+    });
+    expect(mockRouter.push).toHaveBeenCalledWith('/login');
+  });
+
+  it('should show an error message when the token is invalid', async () => {
+    const error = new Error('Invalid token') as Error & { type?: string };
+    error.type = 'user_invalid_token';
+    (resetPassword as jest.Mock).mockRejectedValue(error);
+
+    render(<ResetPassword />);
+    fillFormAndSubmit();
+
+    await waitFor(() => {
+      expect(mockToast).toHaveBeenCalledWith(
+        <Alert
+          variant={AlertVariants.Error}
+          message="Invalid recovery token. Your token may have expired or already been used.  Please try again."
+        />,
+      );
+    });
+
+    act(() => {
+      jest.advanceTimersByTime(2000);
+    });
+    expect(mockRouter.push).toHaveBeenCalledWith('/recover-password');
+  });
+
+  it('should show an error message when the password reset fails', async () => {
+    (resetPassword as jest.Mock).mockRejectedValue(
+      new Error('Error resetting password'),
+    );
+
+    render(<ResetPassword />);
+    fillFormAndSubmit();
+
+    await waitFor(() => {
+      expect(mockToast).toHaveBeenCalledWith(
+        <Alert
+          variant={AlertVariants.Error}
+          message="There was an error resetting your password."
+        />,
+      );
+    });
+
+    act(() => {
+      jest.advanceTimersByTime(2000);
```

Comment:
Why are you setting advanceTimersByTime for this test?

Reply:
because in the page.tsx file I have a timer on the push to the /login route so the user can read the toast.   It waits just as the component would to push to /login route otherwise the test looks for the route to be called with /login before it happens causing it to fail.